### ISSUE TITLE
Flutter 2.5.0 build issues on iOS simulator

### ISSUE THREAD

**Issue by kolotum** [cofu-app/cbl-dart#173]
**bug description**
When we update our existing Flutter project to Flutter 2.5.0 there is a build error when we try to run the app in iOS Simulator.

The same project builds with Flutter 2.2.3.

I think the issue does not occur on real iOS devices or Android.

> ld: warning: directory not found for option '-F/path-to-flutter-project/build/ios/Debug-iphonesimulator/XCFrameworkIntermediates/CouchbaseLite'
> ld: warning: directory not found for option '-F/path-to-flutter-project/build/ios/Debug-iphonesimulator/XCFrameworkIntermediates/CouchbaseLiteDart'
> ld: framework not found CouchbaseLite
> clang: error: linker command failed with exit code 1 (use -v to see invocation)
> note: Using new build system
> note: Building targets in parallel
> note: Planning build
> note: Analyzing workspace
> note: Constructing build description
> note: Build preparation complete
>
> Could not build the application for the simulator.

**Steps to reproduce the behavior:**
1. Update Flutter to 2.5.0
2. flutter clean
3. flutter pub get
4. Try to run app in iOS Simulator from Android Studio 

**Environment:**
cbl: ^0.6.0
cbl_flutter: ^0.4.0+3
Xcode: 12.5.1
Android Studio: 2020.3.1 Patch 2

We pinned the project's Flutter version to 2.2.3 for now.

**Comment by blaugold** [comment]
Hey @kolotum, this looks like an issue related to the `arm64` architecture slice not being available for the iOS Simulator libraries. I'm not sure why this became a problem with Flutter v2.5.0. A workaround is to add `arm64` to the excluded architectures for the iOS Simulator, for all pods:

```ruby
post_install do |installer|
  installer.pods_project.targets.each do |target|
    flutter_additional_ios_build_settings(target)
  end

  # Adds `arm64` as an excluded arch for iOS Simulator, for all pods. Can be removed once all pods either
  # ship the `arm64` architecture or exclude it them self.
  installer.pods_project.targets.each do |target|
    target.build_configurations.each do |config|
      config.build_settings["EXCLUDED_ARCHS[sdk=iphonesimulator*]"] = "$(inherited) arm64"
    end
  end
end
```

**Comment by blaugold** [comment]
Closing this issue because it has already been fixed by 13b66967831d75bf35e80623bee167de3d76ddbd.
This fix is available in `cbl_flutter: >=1.0.0-beta.4`.
